tsl/src/bgw_policy/compression_api.c

```diff
@@ -151,13 +155,35 @@ policy_recompression_proc(PG_FUNCTION_ARGS)
 	PG_RETURN_VOID();
 }
 
+static void
+validate_compress_after_type(Oid partitioning_type, Oid compress_after_type)
+{
+	Oid expected_type = InvalidOid;
+	if (IS_INTEGER_TYPE(partitioning_type))
+	{
+		if (!IS_INTEGER_TYPE(compress_after_type))
+			expected_type = partitioning_type;
+	}
+	else if (compress_after_type != INTERVALOID)
+	{
+		expected_type = INTERVALOID;
+	}
+	if (expected_type != InvalidOid)
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("unsupported compress_after argument type, expected type : %s",
+						format_type_be(expected_type))));
+	}
+}
+
 Datum
 policy_compression_add(PG_FUNCTION_ARGS)
 {
 	NameData application_name;
 	NameData proc_name, proc_schema, owner;
 	int32 job_id;
-	Oid ht_oid = PG_GETARG_OID(0);
+	Oid user_oid = PG_GETARG_OID(0);
```

Comment:
You should add some comments about what this argument can or cannot be.

Reply:
added

Reply:
Can you rename this one to `user_rel_oid` as well?

Reply:
done.